src/main/kotlin/org/jetbrains/research/ml/kotlinAnalysis/PsiProvider.kt

```diff
@@ -0,0 +1,64 @@
+package org.jetbrains.research.ml.kotlinAnalysis
+
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.roots.ProjectRootManager
+import com.intellij.openapi.vfs.VfsUtilCore
+import com.intellij.psi.PsiComment
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiManager
+import com.intellij.psi.util.PsiTreeUtil
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.psi.KtNamedFunction
+import org.jetbrains.kotlin.psi.KtTreeVisitorVoid
+
+class PsiProvider {
+
+    fun extractMethodsFromProject(project: Project): MutableList<KtNamedFunction> {
+        val ktMethods = mutableListOf<KtNamedFunction>()
+        val ktFiles = extractPsiFiles(project)
+        ktFiles.forEach { ktFile ->
+            collectPsiMethods(ktFile).forEach { function ->
+                ktMethods.add(function)
+            }
+        }
+        return ktMethods
+    }
+
+    private fun extractPsiFiles(project: Project): MutableList<KtFile> {
+        val projectPsiFiles = mutableListOf<KtFile>()
+        ProjectRootManager.getInstance(project).contentRoots.mapNotNull { root ->
+            VfsUtilCore.iterateChildrenRecursively(root, null) { virtualFile ->
+                if ((virtualFile.extension != "kt" && virtualFile.extension != "kts") ||
```

Comment:
Please, add a function witch check if the file is kotlin or kotlin script. Also probably you can create enum class for extensions in file util module since it will be useful in this project:
```
enum class Extension(ext: String) {
        KT("kt"),
        KTS("kts")
    }
```

```
fun VirtualFile.isKotlinRelatedFile(): Boolean {
        return this.extension == Extension.KT.ext || this.extension != Extension.KTS.ext
    }
```